app/src/main/kotlin/io/intrepid/skotlinton/base/BaseFragment.kt

```diff
@@ -20,7 +20,10 @@ abstract class BaseFragment<P : BaseContract.Presenter> : Fragment(), BaseContra
         get() = activity.application as SkotlintonApplication
     protected abstract val layoutResourceId: Int
 
-    protected lateinit var presenter: P
+    protected val presenter: P by lazy(LazyThreadSafetyMode.NONE) {
+        val configuration = skotlintonApplication.getPresenterConfiguration()
+        createPresenter(configuration)
+    }
```

Comment:
I'm curious, is there any gain in this? Aren't we instantiating new presenters when the activity/fragment starts and calling `bind()` and `onViewCreated()`?

Reply:
There's no reason the view should ever live longer than the presenter, so there's no reason the presenter references should ever need to change. It's lazy in case anything inside of `createPresenter()` is lifecycle-dependent, as opposed to just creating it at constructor time.